packaging/os/pulp_repo.py

```diff
@@ -0,0 +1,291 @@
+
+    def parse_repo_list(self, repo_list):
+        repo_list_lines = repo_list.split("\n")[4:]
+        all_repos = []
+        repo = {}
+
+        for line in repo_list_lines:
+            # If we have a blank line, we're in between 2 repos
+            if line == '':
+                all_repos.append(repo)
+                repo = {}
+                continue
+            if line.startswith('Id:'):
+                repo['repo_id'] = line.split(":", 1)[1].strip()
+                continue
+            if line.startswith('Display Name:'):
+                repo['display_name'] = line.split(":", 1)[1].strip()
+                continue
+            if line.startswith('Description:'):
+                repo['description'] = line.split(":", 1)[1].strip()
+                continue
+            if line.startswith('    Feed:'):
+                repo['feed_url'] = line.split(":", 1)[1].strip()
+                continue
+            if line.startswith('    Relative URL:'):
+                repo['relative_url'] = line.split(":", 1)[1].strip()
+                continue
+        return all_repos
+
+    def set_pulp_base_command(self):
+        cmd = self.pulp_absolute_path
+        if self.username:
+            cmd += " --username=%s" % pipes.quote(self.username)
+        if self.password:
+            cmd += " --password=%s" % pipes.quote(self.password)
+        cmd += " %s repo" % pipes.quote(self.repo_type)
+        cmd += " "
+        self.pulp_command = cmd
+
+    def set_repo_feed_url(self, repo_id, feed_url):
+        cmd = self.pulp_command
+        cmd += "update "
+        cmd += " --repo-id=%s " % repo_id
+        cmd += " --feed=%s " % feed_url
+        update_output = self.module.run_command(cmd)
+        if update_output[0] != 0:
+            self.module.fail_json(msg="Error setting feed URL on repo.")
+
+    def set_repo_relative_url(self, repo_id, relative_url):
+        cmd = self.pulp_command
+        cmd += "update "
+        cmd += " --repo-id=%s " % repo_id
+        cmd += " --relative-url=%s " % relative_url
+        update_output = self.module.run_command(cmd)
+        if update_output[0] != 0:
+            self.module.fail_json(msg="Error setting relative URL on repo.")
+
+    def set_repo_list(self):
+        cmd = self.pulp_command
+        cmd += "list --details"
+        repo_list_output = self.module.run_command(cmd)
+        if repo_list_output[0] != 0:
+            self.module.fail_json(msg="Error getting repo list.")
+        self.repo_list = self.parse_repo_list(repo_list_output[1])
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True, aliases=['repo']),
+            state=dict(default="present", choices=["absent", "present"]),
+            repo_type=dict(required=True),
+            login_user=dict(),
+            login_password=dict(no_log=True),
+            feed_url=dict(),
+            relative_url=dict()
```

Comment:
Here are missing additional parameters like `proxy_host` and `serve_http` and any other parameter provided by the `pulp-admin rpm repo create --help` command.